### PR TITLE
Enhancement: Reverse proxying the UI

### PR DESCRIPTION
**Pull request by znicholasbrown**
tldr; the UI is now accessible via reverse proxy by setting the `PREFECT_UI_SERVE_BASE` variable (note that this shouldn't be a fully-qualified URL but should instead represent the subdirectory used to access the index file)

This PR makes some minor tweaks to the way the UI is built when using the dev CLI and mounted when running `prefect server start`. Mounting should now respect the `PREFECT_UI_SERVE_BASE` command and the initial settings call (used to fetch the API base url) will no longer use the base url injected from environment variables, which would cause it to ignore sub directories and thereby fail. 

To test the build locally,  one can run the following commands:

```terminal
$ export PREFECT_UI_SERVE_BASE='/hello-world/'
$ prefect dev build-ui
$ prefect server start
```
This will begin to serve the UI at `http://127.0.0.1:4200/hello-world/` (note that there's no redirect when visiting `/`; it should be the job of the reverse proxy to ensure traffic goes there if needed).

To test the dev UI locally, one must set the api url first (without this the UI will send all requests to the dev server base url, which defaults to `http://127.0.0.1:5432`):
```terminal
$ export PREFECT_UI_API_URL="http://127.0.0.1:4200/api"
$ prefect server start
$ cd ui && npm ci && npm run serve
```

Last, this introduces a `--no-install` flag to the `prefect dev build-ui` command to ease local development of the build itself as I found myself needing to run that command a lot without any dependency changes.

### CODE REVIEW

File: ui/src/services/uiSettings.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class UiSettings {
   public static settings: Settings | null = null
 
   private static promise: Promise<Settings> | null = null
-  private static readonly baseUrl = MODE() === 'development' ? 'http://127.0.0.1:4200' : BASE_URL() ?? window.location.origin
+  private static readonly baseUrl = MODE() === 'development' ? 'http://127.0.0.1:4200' : ''
```
Comment (pleek91) [inline ui/src/services/uiSettings.ts:16]: Can you explain this change? Seems odd to have no base url when not in development.

Comment (znicholasbrown) [reply ui/src/services/uiSettings.ts:16]: Yup -
`BASE_URL` is the base url of static assets built when we call `npm run build` and configured in `vite.config.ts`. The `/ui-settings` endpoint is served from the same base route as the UI app itself _minus any subdirectory we mount it to_. It's a somewhat special case because the settings endpoint is served from `/ui-settings` but the UI app is mounted to `/{PREFECT_UI_SERVE_BASE}`. When this is set to `BASE_URL`, the UI incorrectly attempts to fetch settings from the asset base url, which resolves to `/{PREFECT_UI_SERVE_BASE}/ui-settings`

File: src/prefect/server/api/server.py [hunk 2]
```diff
@@ -315,7 +315,7 @@ def ui_settings():
         and not ephemeral
     ):
         ui_app.mount(
-            "/",
+            os.environ.get("PREFECT_UI_SERVE_BASE", "/"),
```
Comment (desertaxle) [inline src/prefect/server/api/server.py:318]: Do we want this to be a setting like `PREFECT_UI_API_URL` so you can configure it in your profile or as an environment variable?

Comment (znicholasbrown) [reply src/prefect/server/api/server.py:318]: Hm I suppose we could do that!

File: ui/src/services/uiSettings.ts (hunk 1 above)
Comment (pleek91) [reply ui/src/services/uiSettings.ts:16]: BASE_URL is also used for the router history. Any change needed there?

File: src/prefect/server/api/server.py (hunk 2 above)
Comment (znicholasbrown) [reply src/prefect/server/api/server.py]: This has been updated @desertaxle

File: ui/src/services/uiSettings.ts (hunk 1 above)
Comment (znicholasbrown) [reply ui/src/services/uiSettings.ts:16]: Sorry, missed this before - shouldn't be any change needed there @pleek91

**Review by pleek91 — APPROVED** [review]
